### PR TITLE
Support for requests during form entry

### PR DESCRIPTION
**Pull request by wpride**
cross-request/context in https://github.com/dimagi/commcare-core/pull/653/files

cc @ctsims @esoergel

### CODE REVIEW

File: src/main/java/application/AbstractBaseController.java [hunk 1]
```diff
@@ -87,6 +88,9 @@
     protected FormplayerRaven raven;
 
     @Autowired
+    protected FormSendCalloutHandler formSendCalloutHandler;
```
Comment (ctsims) [inline src/main/java/application/AbstractBaseController.java]: I think I'm missing why it doesn't make the most sense for the FormSession to not instantiate this directly. It seems like there's a ton of bootstrapping that's being done to pass this down the chain

File: src/main/java/util/PrototypeUtils.java [hunk 2]
```diff
@@ -93,7 +94,8 @@ public static void setupPrototypes(){
                 SeriesData.class.getName(),
                 Graph.class.getName(),
                 XYSeries.class.getName(),
-                BubbleSeries.class.getName()
+                BubbleSeries.class.getName(),
+                SendAction.class.getName(),
```
Comment (ctsims) [inline src/main/java/util/PrototypeUtils.java]: Whoops, this looks like a bug on the commcare side actually, too, if didn't add this.

File: src/main/java/application/AbstractBaseController.java (hunk 1 above)
Comment (wpride) [reply src/main/java/application/AbstractBaseController.java]: Autowiring is more idiomatic Spring and much more feature-ful than direct instantiation (giving us access to the entire set of autowired properties/objects). Ideally we'd autowire the FormSession object as well and then we could skip this long handoff chain - good future refactor

File: src/main/java/util/PrototypeUtils.java (hunk 2 above)
Comment (ctsims) [reply src/main/java/util/PrototypeUtils.java]: Ha, sure enough! https://github.com/dimagi/commcare-android/pull/1851

**Pull request merged**